hal/ia32/ia32.h

```diff
@@ -18,36 +18,37 @@
 #define _HAL_IA32_H_
 
 #include <arch/types.h>
-
 /* io access */
 
 
 static inline u8 hal_inb(void *addr)
 {
 	u8 b;
 
-	__asm__ volatile
-	(" \
-		movl %1, %%edx; \
-		inb %%dx, %%al; \
-		movb %%al, %0;" \
-	:"=b" (b) \
-	:"g" (addr) \
-	:"edx", "eax");
+	/* clang-format off */
+	__asm__ volatile (
+		"movl %1, %%edx\n\t"
+		"inb %%dx, %%al\n\t"
+		"movb %%al, %0"
+	: "=b" (b)
+	: "g" (addr)
+	: "edx", "eax");
```

Comment:
Why use the `ebx` register here.
Same in `hal_out*()` functions.

Reply:
`ebx` is not used here.
If you mean `eax` and `edx`, then they are required by `in` and `dec` instructions.

Reply:
as declared in line 33, line 32 will be converted to:
```c
		movb %al, %bl
```
that is, the `bl` will be used.
Why copy the `al` register to `bl`.

Reply:
Oh, you're right.